common/src/main/java/com/microsoft/identity/common/internal/controllers/ApiDispatcher.java

```diff
@@ -235,6 +240,32 @@ public void run() {
                                         command.getCallback().onCancel();
                                     }
                                 });
+                            } else if (finalException instanceof IntuneAppProtectionPolicyRequiredException) {
+                               if (result.getAuthorizationResult() != null
```

Comment:
This is not implemented in ExceptionAdapter, because the properties needed by MAM sdk cannot be retrieved by the authentication result alone.
1. uid and utid from authrization response
2. upn from token response
3. authority url from request parameter

Please let me know if there is a better place to put this.

Reply:
@rpdome ^

Reply:
apart from authority, it seems like everything else is read from the token result object.

I brought this up because I noticed that the code is being duplicated in two places, and it might be better to consolidate it if possible. (Personally, I'd prefer passing the request param as an optional parameter of .exceptionFromAcquireTokenResult() if that helps consolidating the code, because that means we need to fix it, we can fix it in one place)